### PR TITLE
add file sizes and error reporting to...

### PR DESCRIPTION
**Pull request by nickboldt**
add file sizes and error reporting to prefetch task

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -151,20 +151,28 @@ gulp.task('default', function() {
...
+        reqObjs[currentKey] = request(reqs[currentKey].url, function (err) {
+          if (err) { console.log('DOWNLOADING -> ' + reqs[currentKey].url); console.log(err); console.log(""); }
+        });
+        reqObjs[currentKey].pipe(fs.createWriteStream(path.join(prefetchFolder, currentKey))).on('finish',function() {
+          counter--;
+          if(counter===0) {
+            cb();
+          }
+        });
+        reqObjs[currentKey].on('response', function(res){
```
Comment (dgolovin) [inline gulpfile.js]: From my understanding this 'response' event will happen right before or after 'finish'. This means size will  be printed out with long delay for CDK box. If you want get size upfront you need to do HEAD request for headers only. Then in callback start actual download and track progress in 'data' event probably.

Comment (nickboldt) [inline gulpfile.js]: This spat out console.log() content immediately, as the downloads proceeded. So... there's no delay.

Comment (dgolovin) [inline gulpfile.js]: Because files are tiny. You should try the same trick for cdk box file.

Comment (nickboldt) [inline gulpfile.js]: Started out fine but then...

$➔ gulp prefetch
[11:34:11] Using gulpfile ~\Documents\GitHub\developer-platform-install\gulpfile.js
[11:34:11] Starting 'prefetch'...
DOWNLOADING 57 Kb -> http://cdk-builds.usersys.redhat.com/builds/09-Mar-2016/cdk-2.0.0-beta5.zip
DOWNLOADING 981 Mb -> http://cdk-builds.usersys.redhat.com/builds/09-Mar-2016/rhel-cdk-kubernetes-7.2-21.x86_64.vagrant-virtualbox.box
DOWNLOADING 351 Kb -> http://the.earth.li/~sgtatham/putty/latest/x86/pscp.exe
DOWNLOADING 73 Mb -> http://cdn.azul.com/zulu/bin/zulu8.13.0.5-jdk8.0.72-win_x64.zip
DOWNLOADING 857 Kb -> https://cygwin.com/setup-x86_64.exe
DOWNLOADING 12 Mb -> https://github.com/openshift/origin/releases/download/v1.1.4/openshift-origin-client-tools-v1.1.4-3941102-windows.zip
DOWNLOADING 518 Mb -> https://devstudio.redhat.com/9.0/snapshots/builds/devstudio.product_9.0.mars/latest/all/jboss-devstudio-9.1.0.latest-installer-standalone.jar
buffer.js:389
    throw new Error('toString failed');
    ^

Error: toString failed
    at Buffer.toString (buffer.js:389:11)
    at BufferList.toString (C:\Users\nboldt\Documents\GitHub\developer-platform-install\node_modules\request\node_modules\bl\bl.js:166:33)
    at Request.<anonymous> (C:\Users\nboldt\Documents\GitHub\developer-platform-install\node_modules\request\request.js:1014:32)
    at emitOne (events.js:82:20)
    at Request.emit (events.js:169:7)
    at IncomingMessage.<anonymous> (C:\Users\nboldt\Documents\GitHub\developer-platform-install\node_modules\request\request.js:963:12)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:905:12)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)

*sad trombone*

Comment (nickboldt) [inline gulpfile.js]: Crapped out on rhel-vagrant-virtualbox.box at 695mb of 981mb. Seems there's a limit to the buffer.

**Pull request closed without merging**